Remove `cli/flags` package

vdemeester opened the pull request:
- Moving the `common*.go` files in `cmd/dockerd` directly (it's the
  only place it's getting used)
- Rename `cli/flags` to `cli/config` because it's the only thing left
  in that package 

Now, `integration-cli` does *truly* not depend on `cobra` stuff :sweat_smile:.

It's a follow-up for #33411 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

aaronlehmann reviewed (approved):
LGTM

thaJeztah commented:
Can we rename `common` to something else? The reason it was called `common` is because of;

```
commonOptions are options common to both the client and the daemon.
```

Now that they are only used by `dockerd`, that no longer applies 😅

vdemeester commented:
@thaJeztah you are completely right 👼 :stuck_out_tongue_closed_eyes:

thaJeztah reviewed (approved):
LGTM 🎏

dnephin commented on cmd/dockerd/options.go:
```diff
@@ -27,8 +28,8 @@ var (
-// CommonOptions are options common to both the client and the daemon.
-type CommonOptions struct {
+// daemonFlags are options flags for the daemon (that were share with the client previously).
+type daemonFlags struct {
```
This should probably be `daemonOptions`. They aren't actually flags. The flags are only ever stored on the Command itself. These are just the fields that receive values from flags, which we call options everywhere else.

We already have a `daemonOptions`, so maybe we just merge this struct with the existing `daemonOptions` ?

To reduce the size of the diff maybe just move `LogLevel`, and `Debug` to the existing `daemonOptions` struct, leaving host and tls fields, and this can be `hostOptions` or something like that?

dnephin commented on cmd/dockerd/docker.go:
```diff
@@ -21,14 +20,14 @@ type daemonOptions struct {
 	configFile   string
 	daemonConfig *config.Config
-	common       *cliflags.CommonOptions
+	common       *daemonFlags
```
I guess this `common` field could also use a rename to something else, depending on what name we give the new struct

vdemeester commented:
@dnephin updated 👼

dnephin reviewed (approved):
LGTM

vdemeester commented:
hum failures looks legit.. taking a look

vdemeester committed:
Remove `cli/flags` package

- Moving the `common*.go` files in `cmd/dockerd` directly (it's the
  only place it's getting used)
- Rename `cli/flags` to `cli/config` because it's the only thing left
  in that package 👼

Now, `integration-cli` does *truly* not depend on `cobra` stuff.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester commented:
Updated and :green_heart:

dnephin reviewed (approved):
LGTM

thaJeztah reviewed (approved):
still LGTM
